x-pack/elastic-agent/pkg/agent/application/upgrade/step_download.go

```diff
@@ -20,7 +22,14 @@ import (
 	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/release"
 )
 
-func (u *Upgrader) downloadArtifact(ctx context.Context, version, sourceURI string) (string, error) {
+func (u *Upgrader) downloadArtifact(ctx context.Context, version, sourceURI string) (_ string, err error) {
+	span, ctx := apm.StartSpan(ctx, "upgrade", "app.internal")
```

Comment:
I think the url / filepath passed into the fn will be included in child spans, but it might be nice to include `sourceURI` as a [label on the span](https://pkg.go.dev/go.elastic.co/apm#SpanContext.SetLabel). I'm not sure where best to do this, but maybe up one level, in `*Upgrader.Upgrade()`?

@axw do you have an opinion about @lykkin's question?

Reply:
```suggestion
	span, ctx := apm.StartSpan(ctx, "downloadArtifact", "app.internal")
```

Might be a more informative span name?

@lykkin yes as Stuart points out there's an API for attaching arbitrary labels (strings, booleans, or numbers). These are stored under `labels.<name>` in the resulting span document.

For this particular case I guess the `fetcher.Download` call down below will issue an HTTP request? If so, we should probably instrument that method's internal HTTP client with `apmhttp.WrapClient`. That way we'll see an HTTP client span, and that will have the requested URL and response status code. @stuartnelson3 are we already doing that?

Reply:
Correct, the http client is created [here](https://github.com/elastic/beats/blob/master/x-pack/elastic-agent/pkg/artifact/download/http/downloader.go#L41-L43), which includes wrapping the roundtripper